src/Storage/Commands.Storage/File/Cmdlet/RemoveAzureStorageShare.cs

```diff
@@ -77,9 +91,54 @@ public override void ExecuteCmdlet()
+                    }
+
                     if (force || ShareIsEmpty(share) || ShouldContinue(string.Format("Remove share and all content in it: {0}", share.Name), ""))
                     {
-                        await this.Channel.DeleteShareAsync(share, null, this.RequestOptions, this.OperationContext, this.CmdletCancellationToken).ConfigureAwait(false);
+                        DeleteShareSnapshotsOption deleteShareSnapshotsOption = DeleteShareSnapshotsOption.None;
+                        bool retryDeleteSnapshot = false;
+
+                        //Force means will delete the share anyway, so use 'IncludeSnapshots' to delete the share even has snapshot, or delete will fail when share has snapshot
+                        // To delete a Share shapshot, must use 'None' 
+                        if (IncludeAllSnapshot.IsPresent)
+                        {
+                            deleteShareSnapshotsOption = DeleteShareSnapshotsOption.IncludeSnapshots;
+                        }
+                        else
+                        {
+                            retryDeleteSnapshot = true;
+                        }
+
+                        try
+                        {
+                            await this.Channel.DeleteShareAsync(share, deleteShareSnapshotsOption, null, this.RequestOptions, this.OperationContext, this.CmdletCancellationToken).ConfigureAwait(false);
+                            retryDeleteSnapshot = false;
+                        }
+                        catch (StorageException e)
+                        {
+                            //If x-ms-delete-snapshots is not specified on the request and the share has associated snapshots, the File service returns status code 409 (Conflict).
+                            if (!(e.IsConflictException() && retryDeleteSnapshot))
+                            {
+                                throw;
+                            }
+                        }
+
+                        if (retryDeleteSnapshot)
+                        {
+                            if (force || await OutputStream.ConfirmAsync(string.Format("This share might has snapshots, Remove the share and all snapshots?: {0}", share.Name)).ConfigureAwait(false))
+                            {
+                                deleteShareSnapshotsOption = DeleteShareSnapshotsOption.IncludeSnapshots;
+                                await this.Channel.DeleteShareAsync(share, deleteShareSnapshotsOption, null, this.RequestOptions, this.OperationContext, this.CmdletCancellationToken).ConfigureAwait(false);
+                            }
+                            else
+                            {
+                                string result = string.Format("The remove operation of  share '{0}' is cancelled.", share.Name);
```

Comment:
I think "The remove operation of share '{0}' has been cancelled" makes more sense.  Also remove extra whitespace.

Reply:
Updated.